cloudinit/distros/__init__.py

```diff
@@ -749,6 +752,22 @@ def create_group(self, name, members=None):
                 subp.subp(['usermod', '-a', '-G', name, member])
                 LOG.info("Added user '%s' to group '%s'", member, name)
 
+    def shutdown_command(self, mode='poweroff', delay='now', message=None):
```

Comment:
We always provide values for these, so I wonder if we would be better not specifying defaults here:

```suggestion
    def shutdown_command(self, mode, delay, message):
```

And we are only passing the arguments using keywords, so we could also make them keyword-only:

```suggestion
    def shutdown_command(self, *, mode, delay, message):
```

My thinking here is that we can always convert keyword arguments to positional arguments and/or add defaults in the future if we want: if all callers are passing all arguments as keywords then we can safely do either without affecting existing callers.

What do you think?

Reply:
Yeah. The fact is that there ain't many callers anyway. IDK what the longer run could be for that command. Maybe a default for `delay` or `message` could make sense. But as you say, it's possible to add these defaults later.